### ISSUE TITLE
Theos tweak and swift framework

### ISSUE THREAD

**Issue by ca13ra1** [theos/theos#543]
Is it not possible to include a swift framework in a theos **tweak** not _application_? I’m compiling the swift files and everything works but crashes with unknown error, possibly issues with libswift 5.0. 

[crash log](https://pastebin.com/juYDCs59)

If I remove libswift the tweak doesn’t even get loaded into springboard at all. The swift framework does expose objective c methods as my tweak.xm is in objective c not swift.

Using the latest version of theos. macOS Catalina and Xcode 12. I had this issue back on Xcode 11. Device is on iOS 13.7.

**Comment by kirb** [comment]
Can you check if something related to the crash (e.g. an exception message) shows in syslog?

**Comment by ca13ra1** [comment]
There’s nothing relating to much, that I can see. The code works fine with another framework but it’s outdated so I opted for the swift version, just keeps crashing. 

The code has no problems with the old objective c project.

**Comment by kabiroberai** [comment]
What are the paths of any loaded instances of the libswiftCore.dylib image as reported in the crash log? Is there more than one?

**Comment by ca13ra1** [comment]
Sorry, there's the entire log https://ghostbin.co/paste/kjq3n3s

**Comment by kabiroberai** [comment]
Okay yeah, the issue is that it's loading libswift 5.0 into the process whereas it should be linking against the one in /usr/lib/swift. I believe this should be fixed in the branch `kabiroberai/misc-fixes`; try switching to that branch to see if that fixes it.

**Comment by ca13ra1** [comment]
It worked didn't crash, just not doing anything. The same code works in Xcode, swift converted to objective-c.

**Comment by kabiroberai** [comment]
Does the tweak image get loaded into SpringBoard now? Try checking with FLEX or a similar tool

**Comment by ca13ra1** [comment]
Yes, it's has loaded in SpringBoard.

**Comment by ca13ra1** [comment]
Thanks, I’ll close this issue. Just need to figure out why it’s not initializing.

**Comment by kabiroberai** [comment]
Let's keep this issue open for now until #538 is merged.

**Comment by ca13ra1** [comment]
@kabiroberai seems I can’t get the swift class initialized. It does have exposed objc methods, as I use the same for an Xcode project written it swift which works just fine. Out of ideas, used another swift project and both aren’t actually being initialized. Am I missing something? It compiled the swift files just fine

**Comment by kabiroberai** [comment]
What exactly do you mean by getting the Swift class initialized? Could you give an example? Is there an error or a crash log that you can share?

**Comment by ca13ra1** [comment]
I'm trying to compile a Theos tweak not application using a swift framework. I have made an Xcode using swift framework in objective-c. I modified the Xcode project to fetch images in my tweak directory and it does, successfully so I know the modified swift code works. Then I use the same code in my Theos tweak project and it does absolutely nothing. The ported project to objective-c works but not the swift version.

The tweak class is initialized itself in Flex, but the swift class is never initialized, is this just not possible to use alongside a tweak?
